### PR TITLE
feat: implement channels:activate api adapter and command

### PR DESCRIPTION
**Pull request by oliver-daniel**
Still TODO:
- [x] ~~Implement test~~

### CODE REVIEW

File: src/mix/api/channels.ts [hunk 1]
```diff
@@ -30,3 +30,21 @@ export async function renameChannel(client: MixClient, requestParams: ChannelsRe
...
 }
+
+/**
+ * Activate a channel in a project.
+ * 
+ * @category channels
+ */
+export async function activateChannel(client: MixClient, requestParams: ChannelsActivateParams): Promise<MixResponse> {
+    debug('activateChannel()')
+    const {projectId, channelId} = requestParams
```
Comment (grof) [inline src/mix/api/channels.ts:41]: Use 2-space indenting

File: src/commands/channels/activate.ts [hunk 2]
```diff
@@ -0,0 +1,62 @@
...
+import chalk from 'chalk'
+import {flags} from '@oclif/command'
+import makeDebug from 'debug'
+
+import * as MixFlags from '../../utils/flags'
+import * as ChannelsAPI from '../../mix/api/channels'
+import {DomainOption} from '../../utils/validations'
+import {MixClient, MixResponse} from '../../mix/types'
+import MixCommand from '../../utils/base/mix-command'
+import {ChannelsActivateParams} from '../../mix/api/channels-types'
```
Comment (grof) [inline src/commands/channels/activate.ts:18]: Sort this import alphabetically.

File: src/commands/channels/activate.ts [hunk 3]
```diff
@@ -0,0 +1,62 @@
...
+    '  --channel bc40667c-e0f6-11ec-9d64-0242ac120003',
+  ]
+
+  static flags = {
+    project: MixFlags.projectWithDefaultFlag,
+    channel: {
+      ...MixFlags.channelMultipleFlag, // REVIEW: same as other command
+      multiple: false,
+    },
+    ...MixFlags.machineOutputFlags,
```
Comment (grof) [inline src/commands/channels/activate.ts:38]: Let's use json and yaml flags explicitly at the end of the list of flags and introduced with an "output flags" comment.

File: src/commands/channels/activate.ts [hunk 4]
```diff
@@ -0,0 +1,62 @@
...
+  }
+
+  doRequest(client: MixClient, params: ChannelsActivateParams): Promise<MixResponse> {
+    debug('doRequest()')
+    return ChannelsAPI.activateChannel(client, params)
+  }
+
+  outputHumanReadable(transformedData: any) {
+    debug('outputHumanReadable()')
+    this.log('Channel activated successfully.')
```
Comment (grof) [inline src/commands/channels/activate.ts:60]: How about: "Channel ID cyan(_id_) activated successfully"?

"successfully" is not used consistently in the cli output and probably should have been.

**Comment by grof** [comment]
Needs a few unit tests.

File: src/commands/channels/activate.ts [hunk 5]
```diff
@@ -0,0 +1,64 @@
...
+  Use this command to activate a channel in a project.`
+
+  static examples = [
+    'mix channels:activate -P 1922 \\',
+    '  --channel bc40667c-e0f6-11ec-9d64-0242ac120003',
+  ]
+
+  static flags = {
+    project: MixFlags.projectWithDefaultFlag,
+    channel: flags.string({
```
Comment (grof) [inline src/commands/channels/activate.ts:34]: add "C" as short form to be consistent with other channels commands. Perhaps it's time to for channel have its own entry in flags.ts?

File: src/commands/channels/activate.ts [hunk 6]
```diff
@@ -0,0 +1,64 @@
...
+    const {project: projectId, channel: channelId} = options
+
+    return {projectId, channelId}
+  }
+
+  doRequest(client: MixClient, params: ChannelsActivateParams): Promise<MixResponse> {
+    debug('doRequest()')
+    return ChannelsAPI.activateChannel(client, params)
+  }
+
```
Comment (grof) [inline src/commands/channels/activate.ts:59]: Let's override `setRequestActionMessage` to say:

"Activating channel with ID cyan(channelId)"

File: src/commands/channels/activate.ts (hunk 5 above)
Comment (grof) [reply src/commands/channels/activate.ts:34]: Actually, following our conversation regarding "nice inputs", I suggest we be forward looking. What about we rename this flag to `--channel-id` with no short form.  We then can keep `--channel` with the short form `-C` for when we support using the channel name as unique identifier.  `--channel-id` would still be supported for backward compatibility (and to skip the extra endpoint call).

Comment (oliver-daniel) [reply src/commands/channels/activate.ts:34]: Good point. Once everything is merged into the topic branch, let's take another look and see if we can refactor the flags.

**Comment by grof** [comment]
all good.

**Pull request merged**